tests/integration/vrf_test.yaml

```diff
@@ -0,0 +1,369 @@
+  state: absent
+- name: test-bond0.100
+  type: vlan
+  state: absent
+- name: test-vrf0
+  type: vrf
+  state: absent
+description: Create the vrf interface test-vrf0 with vrf port eth1 and vrf route table ID 100
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth1
+    route-table-id: 100
+description: Delete the vrf interface test-vrf0
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: absent
+description: Delete the vrf interface test-vrf0
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: absent
+description: Create the vrf interface test-vrf0 with vrf port eth1 and vrf route table ID 100
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth1
+    route-table-id: 100
+description: Set the empty port for the vrf interface test-vrf0
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port: []
+description: Set the empty port for the vrf interface test-vrf0
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port: []
+description: Create the vrf interface test-vrf0 with vrf port eth1 and vrf route table ID 100
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth1
+    route-table-id: 100
+description: Delete the vrf interface test-vrf0
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: absent
+description: Create the vrf interface test-vrf1, attach ports eth1 and eth2 to it, configure the vrf route table ID 101
+interfaces:
+- name: test-vrf1
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth1
+    - eth2
+    route-table-id: 101
+description: Delete the vrf interface test-vrf1
+interfaces:
+- name: test-vrf1
+  type: vrf
+  state: absent
+description: Create the vrf interface test-vrf1 with vrf port eth2 and vrf route table ID 101
+interfaces:
+- name: test-vrf1
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth2
+    route-table-id: 101
+description: Delete the vrf interface test-vrf1
+interfaces:
+- name: test-vrf1
+  type: vrf
+  state: absent
+description: Create the vrf interface test-vrf0 with port eth2, create the vrf interface test-vrf1 with port eth1
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth2
+- name: test-vrf1
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth1
+description: Create the vrf interface test-vrf0 with vrf port eth1 and vrf route table ID 100
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth1
+    route-table-id: 100
+description: Delete the vrf interface test-vrf0
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: absent
+description: Delete the vrf interface test-vrf0 and veth device veth0
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: absent
+- name: veth0
+  type: ethernet
+  state: absent
+description: Configure the vrf interface test-vrf1 with ports veth0 and vrf route table ID 101, configure the veth interface veth0 with controller test-vrf1
```

Comment:
Normal does not say `config xxx with controller`, but might use `attach ethernet veth0 to test-vrf1`